// controllers/user.controller.js
const User = require ('../models/user.model');


exports.createUser = function (req, res) {  
    let user = new User(
        {
            nom : req.body.nom ,
            adresse: req.body.adresse ,
            cp: req.body.cp ,
            ville: req.body.ville ,
            contact: req.body.contact ,
            telephone: req.body.telephone ,
            mail: req.body.mail ,
            prospect: req.body.prospect
        }
    );


    user.save((err) =>{
        if(err)
        {
            console.log(err);
        }
        else
        {
             console.log("User created");
            
        }
       res.send(user);
    })
};





exports.getUser = function (req, res) {
    User.findById(req.params.id, function (err, user) {
          if(err)
        {
            console.log(err);
        }
        res.send(user);
    })
};

exports.userDelete = function (req, res) {
    User.findByIdAndRemove(req.params.id, function (err) {
       if(err)
        {
            console.log(err);
        }
         else
        {
             console.log("User Deleted");
            
        }
        
        res.send('User Deleted');
      
    })
};  

exports.userUpdate = function (req, res) {
    User.findOneAndUpdate(req.params.id, {$set: req.body}, function (err, user) {
          if(err)
        {
            console.log(err);
        }
        else
        {
            console.log("User Updated");
            
        }
        
        res.send('User Udpated');

    })
};
